packages/libs/eda/src/lib/core/components/visualizations/implementations/HistogramVisualization.tsx

```diff
@@ -442,6 +451,85 @@ function HistogramViz(props: VisualizationProps<Options>) {
     [options, providedOverlayVariable, providedOverlayVariableDescriptor]
   );
 
+  // get distribution data
+  const subsettingClient = useSubsettingClient();
+
+  const getData = useCallback(async () => {
+    if (vizConfig.xAxisVariable != null && xAxisVariable != null) {
+      const [displayRangeMin, displayRangeMax, binWidth, binUnits] =
+        NumberVariable.is(xAxisVariable)
+          ? [
+              xAxisVariable.distributionDefaults.displayRangeMin ??
+                xAxisVariable.distributionDefaults.rangeMin,
+              xAxisVariable.distributionDefaults.displayRangeMax ??
+                xAxisVariable.distributionDefaults.rangeMax,
+              xAxisVariable.distributionDefaults.binWidth,
+              undefined,
+            ]
+          : [
+              (xAxisVariable as DateVariable).distributionDefaults
+                .displayRangeMin ??
+                (xAxisVariable as DateVariable).distributionDefaults.rangeMin,
+              (xAxisVariable as DateVariable).distributionDefaults
+                .displayRangeMax ??
+                (xAxisVariable as DateVariable).distributionDefaults.rangeMax,
+              (xAxisVariable as DateVariable).distributionDefaults.binWidth,
+              (xAxisVariable as DateVariable).distributionDefaults.binUnits,
+            ];
+
+      const distribution = await getDistribution<DistributionResponse>(
+        {
+          entityId: vizConfig.xAxisVariable?.entityId ?? '',
+          variableId: vizConfig.xAxisVariable?.variableId ?? '',
+          filters: filters,
+        },
+        (filters) => {
+          return subsettingClient.getDistribution(
+            studyMetadata.id,
+            vizConfig.xAxisVariable?.entityId ?? '',
+            vizConfig.xAxisVariable?.variableId ?? '',
+            {
+              valueSpec: 'count',
+              filters,
+              binSpec: {
```

Comment:
Isn't it easier to leave out the `binSpec`?  

It's optional
https://veupathdb.github.io/EdaSubsettingService/api.html#resources:/studies/{study-id}/entities/{entity-id}/variables/{variable-id}/distribution:post

let the back end figure it all out...

Reply:
Thank you for your comment, @bobular  👍 IIRC, I just tested with empty sub-parameters such as binSpec.displayRangeMin etc. that complained, but honestly didn't try to strip out whole binSpec parameter. Will check if it works: yes this will definitely simplify things if doable.

Reply:
I've also just removed the entire `binSpec` object from the request data and things seem to be working the same (and much more readable).